jplag/src/main/java/de/jplag/clustering/ClusteringFactory.java

```diff
@@ -0,0 +1,72 @@
+package de.jplag.clustering;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import de.jplag.JPlagComparison;
+import de.jplag.Submission;
+import de.jplag.clustering.algorithm.AgglomerativeClustering;
+import de.jplag.clustering.algorithm.GenericClusteringAlgorithm;
+import de.jplag.clustering.algorithm.SpectralClustering;
+import de.jplag.clustering.preprocessors.CumulativeDistributionFunctionPreprocessor;
+import de.jplag.clustering.preprocessors.PercentileThresholdProcessor;
+import de.jplag.clustering.preprocessors.ThresholdPreprocessor;
+
+/**
+ * Runs the clustering according to an options object.
+ */
+public class ClusteringFactory {
+    public static List<ClusteringResult<Submission>> getClusterings(Collection<JPlagComparison> comparisons, ClusteringOptions options) {
+        if (!options.isEnabled())
+            return Collections.emptyList();
+
+        // init algorithm
+        GenericClusteringAlgorithm clusteringAlgorithm = null;
+        if (options.getAlgorithm() == ClusteringAlgorithm.AGGLOMERATIVE) {
+            clusteringAlgorithm = new AgglomerativeClustering(options);
+        } else if (options.getAlgorithm() == ClusteringAlgorithm.SPECTRAL) {
+            clusteringAlgorithm = new SpectralClustering(options);
+        }
+
+        // init preprocessor
+        ClusteringPreprocessor preprocessor;
+        switch (options.getPreprocessor()) {
+            case CUMULATIVE_DISTRIBUTION_FUNCTION:
+                preprocessor = new CumulativeDistributionFunctionPreprocessor();
+                break;
+            case THRESHOLD:
+                preprocessor = new ThresholdPreprocessor(options.getPreprocessorThreshold());
+                break;
+            case PERCENTILE:
+                preprocessor = new PercentileThresholdProcessor(options.getPreprocessorPercentile());
+                break;
+            case NONE:
+            default:
+                preprocessor = null;
+                break;
```

Comment:
My honest opinion:

Such switch case statements should be avoided. 

Simply add a attribute of Type `Function<ClusteringOptions, ClusteringPreprocessor>` to the enum values.
This would make it more easy to create new preprocessors later on & you can simply delete the switch-case